android/modules/calendar/src/java/ti/modules/titanium/calendar/RecurrenceRuleProxy.java

```diff
@@ -0,0 +1,372 @@
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Kroll.proxy
+public class RecurrenceRuleProxy extends KrollProxy
+{
+
+	private static final String TAG = "RecurrenceRule";
+
+	// Keys for daysOfTheWeek dictionary.
+	private final String dayOfWeekKey = "daysOfWeek";
+	private final String weekNumberKey = "week";
+
+	// Keys for end dictionary.
+	private final String until = "endDate";
+	private final String count = "occurrenceCount";
+
+	// Field with the Date of the event begin to provide parity with iOS.
+	// Some information exposed can't be get solely from the recurrence rule.
+	private Date eventBegin;
+	private String rRule;
+
+	//region values that can be set from a Creation Dictionary
+	private TiRecurrenceFrequencyType frequency;
+	private Integer interval = 1;
+	private int[] daysOfTheMonth = new int[] {};
+	private int[] daysOfTheYear = new int[] {};
+	private int[] monthsOfTheYear = new int[] {};
+	private int[] weeksOfTheYear = new int[] {};
+	private String calendarID;
+	private KrollDict endDictionary = new KrollDict();
+	private KrollDict[] daysOfTheWeek = new KrollDict[] {};
+	//endregion
+
+	// Map matching days of the week constants from Titanium docs with their String counterparts in RRULE column.
+	private static final Map<String, Integer> weekdaysMap;
+	static
+	{
+		weekdaysMap = new HashMap<String, Integer>();
+		weekdaysMap.put("SU", 1);
+		weekdaysMap.put("MO", 2);
+		weekdaysMap.put("TU", 3);
+		weekdaysMap.put("WE", 4);
+		weekdaysMap.put("TH", 5);
+		weekdaysMap.put("FR", 6);
+		weekdaysMap.put("SA", 7);
+	}
+
+	// Constructor for creating a recurrence rule from a Kroll creation dictionary.
+	public RecurrenceRuleProxy(KrollDict creationDictionary)
+	{
+		if (creationDictionary.containsKey(TiC.PROPERTY_CALENDAR_ID)) {
+			this.calendarID = TiConvert.toString(creationDictionary.get(TiC.PROPERTY_CALENDAR_ID));
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_MONTH)) {
+			this.daysOfTheMonth = creationDictionary.getIntArray(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_MONTH);
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_WEEK)) {
+			this.daysOfTheWeek = creationDictionary.getKrollDictArray(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_WEEK);
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_YEAR)) {
+			this.daysOfTheYear = creationDictionary.getIntArray(TiC.PROPERTY_CALENDAR_DAYS_OF_THE_YEAR);
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_END)) {
+			this.endDictionary = (KrollDict) creationDictionary.getKrollDict(TiC.PROPERTY_END);
+		}
+		if (creationDictionary.containsKey(TiC.PROPERTY_FREQUENCY)) {
+			this.frequency =
+				TiRecurrenceFrequencyType.fromTiIntId(TiConvert.toInt(creationDictionary.get(TiC.PROPERTY_FREQUENCY)));
+		}
```

Comment:
iOS defaults to DAILY if the frequency was not set. We should do the same. Something like this...
```java
if (creationDictionary.containsKey(TiC.PROPERTY_FREQUENCY)) {
	this.frequency = TiRecurrenceFrequencyType.fromTiIntId(TiConvert.toInt(creationDictionary.get(TiC.PROPERTY_FREQUENCY)));
}
if (this.frequency == null) {
	this.frequency = TiRecurrenceFrequencyType.DAILY;
}
```